services/datamanager/builtin/capture/file_count_logger.go

```diff
@@ -0,0 +1,87 @@
+package capture
+
+import (
+	"context"
+	"os"
+	"path/filepath"
+	"time"
+
+	goutils "go.viam.com/utils"
+
+	"go.viam.com/rdk/data"
+	"go.viam.com/rdk/logging"
+	datasync "go.viam.com/rdk/services/datamanager/builtin/sync"
+)
+
+// Threshold number of files to check if sync is backed up (defined as >1000 files).
+var minNumFilesToLog = 1000
+
+// fileCountLogger logs the number of completed capture files that
+// are eligible to be synced in the capture directory.
+// It excludes files in failed directories
+// This is a temporary measure to provide some imperfect signal into what
+// data capture files are eligible to be synced on the robot.
+type fileCountLogger struct {
+	logger logging.Logger
+	worker *goutils.StoppableWorkers
+}
+
+func newFileCountLogger(logger logging.Logger) *fileCountLogger {
+	return &fileCountLogger{
+		logger: logger,
+		worker: goutils.NewBackgroundStoppableWorkers(),
+	}
+}
+
+func (poller *fileCountLogger) reconfigure(captureDir string) {
```

Comment:
This is going to be replaced by the utility you showed in the eng meeting right?

Reply:
yep, already done